src/PowerPing/Ping.cs

```diff
@@ -365,13 +345,18 @@ private PingResults SendICMP(PingAttributes attrs)
             packetSize = packet.messageSize + 4;
 
             // Sending loop
-            while (attrs.Continous ? true : index <= attrs.Count) {
+            for (int index = 1; attrs.Continous || index <= attrs.Count; index++) {
 
-                // Exit loop if cancel event is set
-                if (cancelEvent.WaitOne(0)) {
-                    break;
-                } else {
-                    IsRunning = true;
+                if (index != 1) {
```

Comment:
The wait used to happen unconditionally at the end of the loop. I wanted to eliminate the pause after the final ping and this is the most elegant way to do it in my opinion.

Reply:
Ah ok I understand what you mean, it does make more sense to do it at the start of the loop. 

So why don't we want to run it on 1st run of the loop exactly? Is it because you don't want a delay when sending the first ping?

Reply:
Exactly, we only want to pause in between pings:
Ping 1 -- (Pause) -- Ping 2 -- (Pause) -- Ping 3
So depending where the pause is in the loop, we need to skip it on either the first or last iteration. If I kept the pause at the bottom of the loop, the 'if' logic would have been more complicated:
if (attrs.Continuous || index != attrs.Count) // Pause